server/src/test/java/org/elasticsearch/search/query/QueryPhaseCollectorManagerTests.java

```diff
@@ -0,0 +1,708 @@
+        }
+    }
+
+    public void testPostFilterTopDocsOnly() throws IOException {
+        {
+            CollectorManager<TopScoreDocCollector, TopDocs> topDocsManager = TopScoreDocCollector.createSharedManager(1, null, 1000);
+            TermQuery termQuery = new TermQuery(new Term("field2", "value"));
+            Weight filterWeight = termQuery.createWeight(searcher, ScoreMode.TOP_DOCS, 1f);
+            QueryPhaseCollectorManager<TopDocs, ?> manager = QueryPhaseCollector.createManager(topDocsManager, filterWeight, 0, null, null);
+            QueryPhaseCollectorManager.Result<TopDocs, ?> result = searcher.search(new MatchAllDocsQuery(), manager);
+            assertFalse(result.terminatedAfter());
+            assertEquals(numField2Docs, result.topDocs().totalHits.value);
+        }
+        {
+            CollectorManager<TopScoreDocCollector, TopDocs> topDocsManager = TopScoreDocCollector.createSharedManager(1, null, 1000);
+            TermQuery termQuery = new TermQuery(new Term("field1", "value"));
+            Weight filterWeight = termQuery.createWeight(searcher, ScoreMode.TOP_DOCS, 1f);
+            QueryPhaseCollectorManager<TopDocs, ?> manager = QueryPhaseCollector.createManager(topDocsManager, filterWeight, 0, null, null);
+            QueryPhaseCollectorManager.Result<TopDocs, ?> result = searcher.search(new MatchAllDocsQuery(), manager);
+            assertFalse(result.terminatedAfter());
+            assertEquals(numDocs, result.topDocs().totalHits.value);
+        }
+    }
+
+    public void testPostFilterWithAggs() throws IOException {
+        {
+            CollectorManager<TopScoreDocCollector, TopDocs> topDocsManager = TopScoreDocCollector.createSharedManager(1, null, 1000);
+            CollectorManager<DummyTotalHitCountCollector, Integer> aggsManager = DummyTotalHitCountCollector.createManager();
+            TermQuery termQuery = new TermQuery(new Term("field1", "value"));
+            Weight filterWeight = termQuery.createWeight(searcher, ScoreMode.TOP_DOCS, 1f);
+            QueryPhaseCollectorManager<TopDocs, Integer> manager = QueryPhaseCollector.createManager(
+                topDocsManager,
+                filterWeight,
+                0,
+                aggsManager,
+                null
+            );
+            QueryPhaseCollectorManager.Result<TopDocs, Integer> result = searcher.search(new MatchAllDocsQuery(), manager);
+            assertFalse(result.terminatedAfter());
+            assertEquals(numDocs, result.topDocs().totalHits.value);
+            assertEquals(numDocs, result.aggs().intValue());
+        }
+        {
+            CollectorManager<TopScoreDocCollector, TopDocs> topDocsManager = TopScoreDocCollector.createSharedManager(1, null, 1000);
+            CollectorManager<DummyTotalHitCountCollector, Integer> aggsManager = DummyTotalHitCountCollector.createManager();
+            TermQuery termQuery = new TermQuery(new Term("field2", "value"));
```

Comment:
Let's don't use dummy collectors where we should be using an aggregation specific collector, please. This will be limiting in the long term if we want to do some more specific aggregation stuff here.

Reply:
Is there a lightweight aggregation specific collector that I can use? Can you point me to how I can create one please?

Reply:
I am leaning towards changing this as a follow-up. I would gladly take help on this. I tried it and probably took a wrong direction but it required a non negligible amount of context related objects.

By the way I have attempted to do the same (and failed so far) in `QueryPhaseTests`, as we have many variations of tests there but never aggs in the mix. I would say that if we want real aggs collectors `QueryPhaseTests` would be a good place to start, as it tests the interaction between the different parts of the query phase, while this specific test is testing a specific collector, which I think makes it more suited for dummy/simple wrapped collectors. That said if there is a way to get something more similar to what we have in real-life, which does not have too many dependencies, that seems like a good addition to this test too.

Reply:
Agree, there is not an easy solution as far as we make aggregations look like lucene collectors.